### PR TITLE
Support streaming incoming events when creating app instances

### PR DESCRIPTION
**Pull request by KodrAus**
This PR adds support for creating app instances that receive streamed events using a `--stream` flag. You can supply a signal expression to filter streamed events using the `--stream-signal` flag (which implies `--stream`).

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "sdk": {
-    "version": "7.0.400"
```
Comment (nblumhardt) [inline global.json:3]: Accidental inclusion?

File: src/SeqCli/Cli/Commands/AppInstance/CreateCommand.cs [hunk 2]
```diff
@@ -48,12 +49,16 @@ public CreateCommand(SeqConnectionFactory connectionFactory, SeqCliConfig config
                 _settings.Add(name, valueText ?? "");
             });
 
+        Options.Add(
+            "stream",
+            "Stream incoming events",
+            _ => _streamIncomingEvents = true
```
Comment (nblumhardt) [inline src/SeqCli/Cli/Commands/AppInstance/CreateCommand.cs:55]: Could the argument here be a signal expression? I.e. `--stream` (no argument) would stream all events, `--stream=signal-1,signal-2` would stream the intersection of these two signals.

IIRC there's a `SignalExpressionPart.Parse` method which should make this pretty smooth 🤞 ...

**Comment by nblumhardt** [comment]
```
RUNNING AppInstanceBasicsTestCase                         
Unrecognized options: To=example@example.com
FAIL
Xunit.Sdk.EqualException: Assert.Equal() Failure: Values differ
Expected: 0
Actual:   1
   at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in /_/src/xunit.assert/Asserts/EqualityAsserts.cs:line 251
   at SeqCli.EndToEnd.AppInstance.AppInstanceBasicsTestCase.ExecuteAsync(SeqConnection connection, ILogger logger, CliCommandRunner runner) in /home/appveyor/projects/seqcli/test/SeqCli.EndToEnd/AppInstance/AppInstanceBasicsTestCase.cs:line 25
   at SeqCli.EndToEnd.Support.IsolatedTestCase.ExecuteTestCaseAsync() in /home/appveyor/projects/seqcli/test/SeqCli.EndToEnd/Support/IsolatedTestCase.cs:line 63
   at SeqCli.EndToEnd.Support.TestDriver.Run() in /home/appveyor/projects/seqcli/test/SeqCli.EndToEnd/Support/TestDriver.cs:line 55
```

**Comment by KodrAus** [comment]
This should be fixed up now :+1: Did we want to take a closer look at the arguments themselves? I've currently got:

- `--stream`: Stream all events.
- `--stream-signal`: Specify the signal expression.

It might be better to align these names to the API properties they affect:

- `--stream-incoming-events`
- `--streamed-signal-expression`

That's pretty wordy, but makes decisions around parameter naming easier and less arbitrary. What do you think?

**Comment by nblumhardt** [comment]
I can't help feeling like there's only one argument here; `--stream` implies `--stream-signal` with an empty signal expression, and `--stream-signal` implies `--stream`.

Perhaps just `--stream` with an optional argument?

**Comment by KodrAus** [comment]
> Perhaps just --stream with an optional argument?

I would like to do this too, but wasn't sure what the syntax in our argument parser is for optional values.

**Comment by nblumhardt** [comment]
The parser is a fork of NDesk.Options/Mono.Options; this post suggests it should pretty much "just work" - https://stackoverflow.com/questions/29402914/setting-optional-values-using-ndesk-options - I'm not sure whether the use of `:` there (in place of `=`) is significant, though.

File: src/SeqCli/Cli/Commands/AppInstance/CreateCommand.cs [hunk 3]
```diff
@@ -51,23 +51,19 @@ public CreateCommand(SeqConnectionFactory connectionFactory, SeqCliConfig config
             });
 
         Options.Add(
-            "stream",
-            "Stream incoming events to this app instance as they're ingested",
-            _ =>
+            "stream:",
+            "Stream incoming events to this app instance as they're ingested. Optionally specify a signal expression to limit the events streamed to just those matching the expression",
+            s =>
```
Comment (nblumhardt) [inline src/SeqCli/Cli/Commands/AppInstance/CreateCommand.cs:56]: Nit, none of the argument help is currently written in sentences, perhaps:

> "Stream incoming events to this app instance as they're ingested; optionally accepts a signal expression specifying which events should be streamed"

**Review by nblumhardt — APPROVED** [review]

**Pull request merged**